jicoco-kotlin/src/main/kotlin/org/jitsi/health/AbstractHealthCheckService.kt

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright @ 2018 - present 8x8, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jitsi.health
+
+import org.jitsi.utils.concurrent.PeriodicRunnable
+import org.jitsi.utils.concurrent.RecurringRunnableExecutor
+import org.jitsi.utils.logging2.Logger
+import org.jitsi.utils.logging2.LoggerImpl
+import org.osgi.framework.BundleActivator
+import org.osgi.framework.BundleContext
+import java.time.Clock
+import java.time.Duration
+import java.time.Instant
+
+abstract class AbstractHealthCheckService(
+    val interval: Duration = Duration.ofSeconds(10),
+    /**
+     * If no health checks have been performed in the last {@code timeout}
+     * period, the service is considered unhealthy.
+     */
+    val timeout: Duration = Duration.ofSeconds(30),
+    /**
+     * The maximum duration that a call to {@link #performCheck()} is allowed
+     * to take. If a call takes longer, the service is considered unhealthy. A
+     * value of {@code null} indicates no max check duration.
+     * <p>
+     * Note that if a check never completes, we rely on {@link #timeout} instead.
+     */
+    private val maxCheckDuration: Duration = Duration.ofSeconds(3),
+    /**
+     * If set, a single health check failure after the initial
+     * {@link #STICKY_FAILURES_GRACE_PERIOD}, will be result in the service
+     * being permanently unhealthy.
+     */
+    private val stickyFailures: Boolean = false,
+    /**
+     * Failures in this period (since the start of the service) are not sticky.
+     */
+    private val stickyFailuresGracePeriod: Duration = stickyFailuresGracePeriodDefault
+): BundleActivator, HealthCheckService, PeriodicRunnable(interval.toMillis())
+{
+    @JvmOverloads
```

Comment:
You can define the primary constructor differently to be able to put `@JvmOverloads` there so we don't need a second one:
```
abstract class AbstractHealthCheckService @JvmOverlods constructor(...
```